flybywiresim/a32nx
# feat(efb): mcdu keyboard input settings

Fixes #5727

## Summary of Changes

* Moved the setting for MCDU keyboard input to the EFB
  * Allows keyboard input to be toggled on for the MCDU
  * Customisable timeout duration (5-120seconds) before inactivity returns the focus to the camera from the MCDU. 
* Removed the requirement for a full reload of the plane to use this - Trade-off is that the MCDU screen is now always considered a touchscreen input by the sim even when the setting is disabled.

Note: May need a change to the relevant docs for this feature as well

## Screenshots (if necessary)


![image](https://user-images.githubusercontent.com/15316958/132089382-dae8c7f0-daa1-4589-91b5-2bc4c69264f7.png)

![image](https://user-images.githubusercontent.com/15316958/132089393-c24c9d09-60eb-419a-bd1a-89bdf5403e15.png)


## References





## Additional context



Discord username (if different from GitHub): 2Cas#1022

## Testing instructions

- Spawn in, turn on EFB, Go to EFB Settings > Sim Options
- Enable keyboard input
- Click anywhere on the MCDU screen, the header should highlight cyan, the camera controls will now be locked and the keyboard input will be captured by the MCDU scratchpad.
- Click anywhere on the MCDU to undo or press alt or ctrl+z or wait for the timeout (default 60s) to finish.
- Disable keyboard input
- MCDU should no longer accept input when clicked
- Check that this setting persists during and after flights. This setting should persist indefinitely once set. 

Known Issues:
- ~~Setting a value outside of the 5-120 second range will appear to accept on the input field. This is not saved, when you switch from one EFB setting to another, this will reset back to the last accepted time (between 5 and 120 seconds)~~


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
